docs/api/apiv3/paths/memberships_form.yml

```diff
@@ -1,34 +1,51 @@
 # /api/v3/memberships/form
 ---
 post:
+  summary: Form create membership
+  operationId: form_create_membership
+  tags:
+    - Memberships
+  description: |-
+    Requests and validates the creation form for memberships. The request payload, if sent, is validated. The form
+    endpoint itself does not create a membership.
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: '../components/schemas/membership_write_model.yml'
+        examples:
+          'global role':
+            $ref: '../components/examples/membership-create-request-global-role.yml'
+          'no notification':
+            $ref: '../components/examples/membership-create-request-no-notification.yml'
+          'custom message':
+            $ref: '../components/examples/membership-create-request-custom-message.yml'
   responses:
     '200':
       description: OK
-      headers: {}
+      content:
+        application/hal+json:
+          schema:
+            $ref: '../components/schemas/membership_form_model.yml'
+          examples:
+            form:
+              $ref: '../components/examples/membership-form-response.yml'
```

Comment:
that is a very good hint, and it took me a while to figure out, why I prefer `examples` over `example` in a [Media Type Object](https://spec.openapis.org/oas/latest.html#mediaTypeObject). The answer is a bit hidden inside this link.

`example` is of type `any`. Meaning, if I want to extract and reference it, the extracted object would no longer be an object of type [Example Object](https://spec.openapis.org/oas/latest.html#exampleObject).

So, my approach here is this:
- use `example` if I would inline any example object, as we are doing it for some error objects
- use `examples` if we extract an example object into a separate file

regarding the explicit naming: I admit, this is born out of laziness, but there is a truth hidden behind that. A explicit text could turn into a misleading, or wrong, description if one changes the example. This cannot happen in a generic text. This is, why I would tend to use explicit namings only to distinguish between multiple examples, because only then the reader of the spec will have a dropdown with those names for toggling the shown example.

What do you think about this?

Reply:
I don't mind about preferring `example` or `examples`. I saw somewhere else that you turned a `examples` into a `example`, so I thought that was a pattern you wanted to reproduce. Now I understand that it was isolated and simple enough and so inlining was better.

About naming examples, I think it is highly useful to understand what is being done. At first when reading the example, I thought that it was wrong because there were some validation errors and it was returning a 200 OK. I thought it was actually a "misleading, or wrong example" but then I read more carefully (and tried the endpoint locally with invalid data) and it is correct.

I think a good descriptive title helps the reader. That the goal of a doc. The "can-be-outdated" reason is also valid for the endpoint description, the endpoint summary, the error code explanations and so on. I don't think that's a good reason to avoid adding them. They provide value for the reader.

In fact, I started appreciating descriptive example names with the 3 different examples you introduced in the patch endpoint here: https://github.com/opf/openproject/pull/14020/files#diff-17a4cc3645bd5b987b7b6b3cb538a7d5dcde99a95ab603fafcfbf1055951c532R122-R128

Reply:
But it's true that when there is only one example in an `examples`, the name is not shown in the rendered api documentation on the website or locally on dev instance. So the name does not really matter as it is not visible from the reader. I assumed it was.

If the name is not visible, then yes, no need to bother with it. Still that would be useful to have an example description for the reader instead of trying to infer what the example is trying to demonstrate.

Reply:
This can also be achieved by added a `description` property to the example object itself.

Reply:
ah good to know.